source/expressions.tex

```diff
@@ -471,15 +471,14 @@
 Using an overloaded operator causes a function call; the
 above covers only operators with built-in meaning.
 \end{note}
-If the (possibly converted) expression is a prvalue,
-the temporary materialization conversion\iref{conv.rval} is applied.
+The temporary materialization conversion\iref{conv.rval} is applied
+if the (possibly converted) expression is a prvalue of object type.
```

Comment:
Does this affect whether a discarded prvalue is materialized?

E.g. with `T f();`, does `static_cast<void>(f());` materialize a `T` object?

Reply:
Yes. "Expression" above is discarded-value expression, and this is part of its definition.

Reply:
Ah I see, this is still talking about the inner expression, not the entire cast expression.